From 5e52d5761fe6c37a0d2318d3e76d55c00d025571 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Thu, 23 Aug 2012 15:52:17 +0530
Subject: [PATCH 24/56] Enabling AAC-ELD decoding support in CIP Codecs.

BZ: 49870

Add AAC-ELD Decoding capability - In CIP Codecs

This fix provides capability to invoke Google-AAC-ELD decoding ONLY
for AAC-ELD profiles and invoke the default Intel MDP optimized AAC
decoding for other profiles.

Category: feature differentiation
Domain: AUDIO_MEDIA-Audio Codecs
Origin: internal
Upstream-Candidate: no, proprietary

Change-Id: Ib4a855ab748b35722dabbb6f8936a8a053f17fab
Orig-Change-Id: I021b8f1b2417e769b7cd28c0a91b1348f7644123
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 include/media/stagefright/MediaDefs.h   | 4 +++-
 media/libstagefright/MPEG4Extractor.cpp | 4 ++++
 media/libstagefright/MediaDefs.cpp      | 4 +++-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/include/media/stagefright/MediaDefs.h b/include/media/stagefright/MediaDefs.h
index 481972c..b3bf98c 100644
--- a/include/media/stagefright/MediaDefs.h
+++ b/include/media/stagefright/MediaDefs.h
@@ -99,7 +99,9 @@ extern const char *MEDIA_MIMETYPE_CONTAINER_WMV;
 extern const char *MEDIA_MIMETYPE_CONTAINER_VC1;
 extern const char *MEDIA_MIMETYPE_CONTAINER_HEVC;
 extern const char *MEDIA_MIMETYPE_CONTAINER_FFMPEG;
-
+#ifdef USE_INTEL_MDP
+extern const char *MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED;
+#endif
 }  // namespace android
 
 #endif  // MEDIA_DEFS_H_
diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 325311e..0eaf845 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -2593,7 +2593,11 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
 
     if (objectType == 31) {  // AAC-ELD => additional 6 bits
         objectType = 32 + br.getBits(6);
+#ifdef USE_INTEL_MDP
+        mLastTrack->meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED);
+#endif
     }
+    mLastTrack->meta->setInt32(kKeyAACProfile, objectType);
 
 #ifdef QCOM_DIRECTTRACK
     if(objectType == 1) { //AAC Main profile
diff --git a/media/libstagefright/MediaDefs.cpp b/media/libstagefright/MediaDefs.cpp
index 6a69740..6104c2c 100644
--- a/media/libstagefright/MediaDefs.cpp
+++ b/media/libstagefright/MediaDefs.cpp
@@ -94,5 +94,7 @@ const char *MEDIA_MIMETYPE_CONTAINER_WMV = "video/x-ms-wmv";
 const char *MEDIA_MIMETYPE_CONTAINER_VC1 = "video/vc1";
 const char *MEDIA_MIMETYPE_CONTAINER_HEVC = "video/hevc";
 const char *MEDIA_MIMETYPE_CONTAINER_FFMPEG = "video/ffmpeg";
-
+#ifdef USE_INTEL_MDP
+const char *MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED = "audio/mp4a-aacextended"; //For Audio Object Type > 31
+#endif
 }  // namespace android
-- 
2.7.4

